From cafdaacb62deaf29cccbbd0b9bad4a32c9a4bef0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 15:17:09 +0200
Subject: [PATCH] [SW-0] pll: replace magic values with enum

Introduce hl_pll_bypass_val enum for the bypass register values.

Change-Id: I7a39306862672bac9a04844bae58fbeec6bc5eaf
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index 68940f0..d40fe16 100644
--- a/pll.c
+++ b/pll.c
@@ -36,6 +36,13 @@
 
 static int cpu_pll_bypassed = 0;
 
+enum hl_pll_bypass_val {
+	HL_PLL_REF_CLK,
+	HL_PLL_PLL_CLK,
+	HL_PLL_DIV_REF,
+	HL_PLL_DIV_PLL,
+};
+
 struct hl_pll_cfg {
 	char	*name;
 	void	*base;
@@ -144,9 +151,9 @@
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
 
 		if (bypass)
-			writel(0, reg);
+			writel(HL_PLL_REF_CLK, reg);
 		else
-			writel(cfg->div_factor[sel] ? 3 : 1, reg);
+			writel(cfg->div_factor[sel] ? HL_PLL_DIV_PLL : HL_PLL_PLL_CLK, reg);
 	}
 
 	/* just to be on a safe side, wait for bypass to propagate */
